lib/Zonemaster/Backend/Config.pm

```diff
@@ -29,29 +30,153 @@ Readonly my @SIG_NAME => split ' ', $Config{sig_name};
+          or die "Repeated locale tag in LANGUAGE.locale: $locale_tag\n";
+
+        $obj->{_LANGUAGE_locale}{$locale_tag} = 1;
+    }
 
-    $self->{cfg} = Config::IniFiles->new( -file => $path );
-    die "UNABLE TO LOAD $path ERRORS:[".join('; ', @Config::IniFiles::errors)."] \n" unless ( $self->{cfg} );
-    bless( $self, $class );
+    $obj->{_public_profiles} = {
+        default => '',
+    };
+    for my $name ( $ini->Parameters( 'PUBLIC PROFILES' ) ) {
+        $obj->{_public_profiles}{lc $name} = $ini->val( 'PUBLIC PROFILES', $name );
+    }
+    $obj->{_private_profiles} = {};
+    for my $name ( $ini->Parameters( 'PRIVATE PROFILES' ) ) {
+        $obj->{_private_profiles}{lc $name} = $ini->val( 'PRIVATE PROFILES', $name );
+    }
 
-    if ( defined $self->{cfg}->val( 'DB', 'database_host' ) ) {
+    # Handle deprecated properties
+    if ( defined( my $value = $ini->val( 'DB', 'database_host' ) ) ) {
         $log->warning( "Use of deprecated config property DB.database_host. Use MYSQL.host or POSTGRESQL.host instead." );
+
+        $obj->{_MYSQL_host} = $value
+          if $obj->BackendDBType eq 'MySQL' && !defined $obj->MYSQL_host;
+
+        $obj->{_POSTGRESQL_host} = $value
+          if $obj->BackendDBType eq 'PostgreSQL' && !defined $obj->POSTGRESQL_host;
     }
-    if ( defined $self->{cfg}->val( 'DB', 'user' ) ) {
+    if ( defined( my $value = $ini->val( 'DB', 'user' ) ) ) {
         $log->warning( "Use of deprecated config property DB.user. Use MYSQL.user or POSTGRESQL.user instead." );
+
+        $obj->{_MYSQL_user} = $value
+          if $obj->BackendDBType eq 'MySQL' && !defined $obj->MYSQL_user;
+
+        $obj->{_POSTGRESQL_user} = $value
+          if $obj->BackendDBType eq 'PostgreSQL' && !defined $obj->POSTGRESQL_user;
     }
-    if ( defined $self->{cfg}->val( 'DB', 'password' ) ) {
+    if ( defined( my $value = $ini->val( 'DB', 'password' ) ) ) {
         $log->warning( "Use of deprecated config property DB.password. Use MYSQL.password or POSTGRESQL.password instead." );
+
+        $obj->{_MYSQL_password} = $value
+          if $obj->BackendDBType eq 'MySQL' && !defined $obj->MYSQL_password;
+
+        $obj->{_POSTGRESQL_password} = $value
+          if $obj->BackendDBType eq 'PostgreSQL' && !defined $obj->POSTGRESQL_password;
     }
-    if ( defined $self->{cfg}->val( 'DB', 'database_name' ) ) {
+    if ( defined( my $value = $ini->val( 'DB', 'database_name' ) ) ) {
         $log->warning( "Use of deprecated config property DB.database_name. Use MYSQL.database, POSTGRESQL.database or SQLITE.database_file instead." );
+
+        $obj->{_MYSQL_database} = $value
+          if $obj->BackendDBType eq 'MySQL' && !defined $obj->MYSQL_database;
+
+        $obj->{_POSTGRESQL_database} = $value
+          if $obj->BackendDBType eq 'PostgreSQL' && !defined $obj->POSTGRESQL_database;
+
+        $obj->{_SQLITE_database_file} = $value
+          if $obj->BackendDBType eq 'SQLite' && !defined $obj->SQLITE_database_file;
```

Comment:
Is this correct or should it be
```
&& !defined $obj->{_SQLITE_database_file};
```

Reply:
One is equivalent to the other. I think it's nicer maintenance-wise to call the method because that way we're limiting the number of places where the private field name is used.